src/commands/aksCRUDDiagnostics/aksCRUDDiagnostics.ts

```diff
@@ -39,19 +39,27 @@ async function loadDetector(
     await longRunning(`Loading ${clustername} diagnostics.`,
           async () => {
             const clusterAppLensData = await getAppLensDetectorData(cloudTarget, "aks-category-crud");
-            const detectorMap = new Map();
+            if (clusterAppLensData === undefined) {
+              vscode.window.showErrorMessage('Unable to retrieve AppLens data.');
+              return;
+            }
 
             // Crud detector list is guranteed form the ARM call to aks-category-crud, under below data structure.
-            const crudDetectorList = clusterAppLensData?.properties.dataset[0].renderingProperties.detectorIds;
-
-            await Promise.all(crudDetectorList.map(async (detector: string) => {
-              const detectorAppLensData = await getAppLensDetectorData(cloudTarget, detector);
-              detectorMap.set(detector , detectorAppLensData);
-            }));
+            const crudDetectorList: string[] = clusterAppLensData?.properties.dataset[0].renderingProperties.detectorIds;
+            if (crudDetectorList.length === 0) {
+              vscode.window.showErrorMessage('No detectors found in AppLens response.');
+              return;
+            }
 
-            if (clusterAppLensData && detectorMap.size > 0) {
-              await createDetectorWebView(clustername, clusterAppLensData, detectorMap, extensionPath);
+            let results: (AppLensARMResponse | undefined)[] = [];
+            try {
+              results = await Promise.all(crudDetectorList.map((detector) => getAppLensDetectorData(cloudTarget, detector)));
+            } catch (err) {
```

Comment:
Adding some error handling. It's possible that the `longRunning` wrapper does some error handling, but if we can handle it closer to the source that's presumably better (and we can at least show an error message here).